presto-main/src/main/java/com/facebook/presto/server/ResourceUtil.java

```diff
@@ -123,6 +126,15 @@ else if (nameParts.size() == 2) {
 
         String value = nameValue.get(1);
 
+        // validate session property value
+        SessionPropertyMetadata<?> metadata = sessionPropertyManager.getSessionPropertyMetadata(fullPropertyName);
+        try {
+            sessionPropertyManager.decodeProperty(fullPropertyName, value, metadata.getJavaType());
+        }
+        catch (Exception e) {
```

Comment:
I think this should be `RuntimeException` since `decodeProperty()` explicitly handles `Exception`